.github/actions/build-connector/action.yml

```diff
@@ -0,0 +1,58 @@
+name: 'Build Power BI Connector'
+description: 'Compiles the Power BI connector and uploads it as an artifact'
+inputs:
+  sdk-path:
+    description: 'Path to the PowerQuery SDK directory'
+    required: true
+  working-directory:
+    description: 'Working directory where the connector project is located'
+    required: false
+    default: '.'
+  artifact-name:
+    description: 'Name for the uploaded artifact'
+    required: false
+    default: 'firebolt-connector'
+outputs:
+  connector-path:
+    description: 'Path to the compiled connector file'
+    value: ${{ steps.compile.outputs.connector-path }}
+  artifact-name:
+    description: 'Name of the uploaded artifact'
+    value: ${{ inputs.artifact-name }}
+runs:
+  using: 'composite'
+  steps:
+    - name: Compile Power BI Connector
+      id: compile
+      shell: powershell
+      working-directory: ${{ inputs.working-directory }}
+      run: |
+        Write-Host "Compiling Power BI Connector..."
```

Comment:
nit: Do we want to put this into a separate script file and just call it? This way it's also reusable